src/Frontend/Core.hs

```diff
@@ -416,23 +416,24 @@ form formHandler = getH :<|> postH
 
 -- | Wrap anything that has 'ToHtml' and wrap it in an HTML body with complete page.
 data Frame body
-    = Frame { _frameUser :: User, _frameBody :: body }
+    = Frame { _frameUser :: User, _frameBody :: body, _frameMessages :: [StatusMessage] }
     | PublicFrame               { _frameBody :: body }
   deriving (Show, Read, Functor)
 
 makeFrame :: (ActionPersist m, ActionUserHandler m, MonadError ActionExcept m, Page p)
           => m p -> m (Frame p)
 makeFrame mp = do
   isli <- isLoggedIn
+  msgs <- flushMessages
   let isPrivate = isPrivatePage mp -- Here 'm' is used as the 'proxy'.
   if | not isli && isPrivate -> redirect . absoluteUriPath $ relPath P.Login
-     | isli     || isPrivate -> Frame <$> currentUser <*> mp
+     | isli     || isPrivate -> Frame <$> currentUser <*> mp <*> pure msgs
```

Comment:
...  or perhaps we should do something: this means that logout flushes the message queue without showing it.  Ok, but not ideal.